### DISCUSSION TITLE
Feature Request: Support PL/SQL and stored procedure

### DISCUSSION THREAD

**Discussion by louishust** [duckdb/duckdb#8104] [Ideas]
Why do you want this feature?

Migrate oracle to duckdb, need migrate the stored procedure.
So if duckdb will support pl/sql and stored procedure?

**Comment by AliHmaou** [comment]
Now it's been 2 years since this idea came up. Duckdb begins to enter deeply in lots of data processings. With flockmtl it's insanely easy to batch run llm powered use cases.
Procédural code support might unleash a whole lot of capabilities directly inside duckdb without having the pain of low efficiency langages such as python.

**Comment by sirinath** [comment]
It will be great if you can use this PL/SQL dialect to create workflows like:
- project a table data into a vector / matrix / tensor by using a pivot table mechanism
- do linear algebra on the metricise
- Un pivot and collapse the resulting matrix back into a table
